src/ert/_c_wrappers/job_queue/forward_model.py

```diff
@@ -63,21 +65,106 @@ def formatted_fprintf(
         run_id,
         path,
         data_root,
-        global_args: SubstitutionList,
+        iens: int,
+        itr: int,
+        substituter: "Substituter",
         env_varlist: EnvironmentVarlist,
+        filename: str = "jobs.json",
     ):
-        self._formatted_fprintf(run_id, path, data_root, global_args, env_varlist)
+        def substitute(job, string):
+            if string is not None:
+                return substituter.substitute(
+                    job.get_private_args().substitute(string), iens, itr
+                )
+            else:
+                return string
+
+        def nulled_int(integer: int) -> Optional[int]:
+            return integer if integer > 0 else None
+
+        def handle_default(job, arg: str) -> str:
+            defaults = job.get_default_mapping()
+            if arg in job.get_default_mapping():
+                return defaults[arg]
+            else:
+                return arg
+
+        def filter_env_dict(job, d):
+            result = {}
+            for key, value in d.items():
+                new_key = substitute(job, key)
+                new_value = substitute(job, value)
+                if new_value is None:
+                    result[new_key] = None
+                elif not (new_value[0] == "<" and new_value[-1] == ">"):
+                    # Remove values containing "<XXX>". These are expected to be
+                    # replaced by substitute, but where not.
```

Comment:
"but were not" instead of "but where not"

Reply:
Fixed